src/semaphore.c

```diff
@@ -474,6 +507,11 @@ _dispatch_group_wake(dispatch_semaphore_t dsema)
 			int ret = sem_post(&dsema->dsema_sem);
 			DISPATCH_SEMAPHORE_VERIFY_RET(ret);
 		} while (--rval);
+#elif USE_FUTEX_SEM
+		do {
+			int ret = _dispatch_futex_signal(&dsema->dsema_futex);
```

Comment:
this is FUTEX_WAIT(rval), do not loop, futexes support the proper interface already